### DISCUSSION TITLE
Unable to reference local images

### DISCUSSION THREAD

**Discussion by cs-mshah** [HEIGE-PCloud/DoIt#970] [Q&A]
I tried to reference an image in the folder containing the post but it didn't render. Is there any reason for this?

This is the directory structure:
```shell
content/
	posts/
		blog1/
			blog1_site.md
			network.png
		blog2/
			blog2_site.md
```

In `blog1_site.md`, when I reference the image using `![](network.png "network")`, it doesn't render even though the image is a page resource. Am I missing something?

**Comment by cs-mshah** [comment]
@HEIGE-PCloud   
can you help?

**Comment by zc-devs** [comment]
Try to use **extended** version of hugo.

**Reply by cs-mshah** [reply]
Here is what my `snap list` shows: `hugo                             0.113.0                     16026  extended/stable`.

I believe I am already using extended. Just try to see if you are able to reference the image as specified on your side.

**Reply by zc-devs** [reply]
I guess it's not a bundle. Try to rename `blog1_site.md` to `index.md`.

When I tested your case I should have rerun Hugo. There were some caching issues...

In the end... Standard version works too. I use config below:
```
$ ls content/posts/docker-in-hyperv-error/
docker-hyperv-unexpected-error.png  index.md
```
```
![An unexpected error was encountered while executing a WSL command](docker-hyperv-unexpected-error.png "Docker in Hyper-V error")
```

**Reply by cs-mshah** [reply]
The thing is I don't want to make it into `index.md`. The reason is that this would make the blog urls to be that of the folder name in which the blog is, with all markdown files with the same `index.md` title. This wouldn't really be great when using a CMS like obsidian for searching the particular md file. I found that creating folders and md files in them would be a good way to organise blogs and their resources (images etc), if it works, but now it seems that the images aren't being referenced, which means that it is pointless to arrange in folders.  

The hugo docs say that images can be referenced relative to the md file. So I don't know why this isn't working. I think I should raise this issue on hugo.

**Reply by zc-devs** [reply]
> why this isn't working
> I don't want to make it into index.md

Because it's not a [bundle](https://gohugo.io/content-management/page-bundles/).

> The hugo docs say that images can be referenced relative to the md file

They can, use `![](../network.png "network")`. Your post is rendered under `http://localhost:1313/posts/blog1/blog1_site/`, but image is under `blog1`.

**Reply by cs-mshah** [reply]
Actually it is rendered under `http://localhost:1313/blog1_site/`. So the folder isn't a part of it. See the live site: [My Blog](https://cs-mshah.github.io/cs161_computer_security_notes/). Here is the structure: [Github](https://github.com/cs-mshah/cs-mshah.github.io/tree/main/content/posts)
